### PR TITLE
Added metadata for org.eclipse.jgit:org.eclipse.jgit:6.5.0.202303070854-r

### PR DESCRIPTION
**Pull request by msupic**
## What does this PR do?
Adds metadata for `org.eclipse.jgit:org.eclipse.jgit:6.5.0.202303070854-r`

## Code sections where the PR accesses files, network, docker or some external service

- The `OrgEclipseJGitTest` creates two git repositories in the `tmp` folder and deletes it at the end of the test execution.

## Checklist before merging
- [x] I have considered including reachability metadata directly in the library or the framework (see [our contributing document](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md))
- [x] I am the original author of all content provided in the pull request, and I did not copy the content from any other source (see [the tests section from our contributing document](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#tests))
- [x] For all tests where I am not the sole author, I have added a comment that proves I may publish them under the specified license (see [the tests section from our contributing document](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#tests))
- [x] I have properly formatted metadata files (see [our formatting guide](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#format-metadata-files))
- [x] I have added thorough tests (see [the tests section from our contributing document](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#Tests))
- [x] I have filled all places where my pull request accesses files, network, docker, or any other external service (see the section above)

### CODE REVIEW

File: tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/build.gradle [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    testImplementation 'ch.qos.logback:logback-classic:1.4.5'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "metadata-user-code-filter.json"
+                extraFilterPath = "metadata-extra-filter.json"
```
Comment (olpaw) [inline tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/build.gradle:26]: Why is this needed? Why does `userCodeFilterPath` alone not work?

File: tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/gradle.properties [hunk 2]
```diff
@@ -0,0 +1,2 @@
+library.version = 6.5.0.202303070854-r
+metadata.dir = org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/
```
Comment (olpaw) [inline tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/gradle.properties:2]: `metadata.dir` can be removed since it is not used in `build.gradle`?

File: tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/src/test/java/org_eclipse_jgit/org_eclipse_jgit/OrgEclipseJGitTest.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        remoteRepositoryDir = new File(TMP_DIR, "test_jgit_remote_repo_" + System.currentTimeMillis());
+        remoteRepository = new FileRepository(new File(remoteRepositoryDir, GIT_DIR_NAME));
+        remoteRepository.create();
+        logger.info("Created remote repository: {}", remoteRepositoryDir);
+        remoteGit = new Git(remoteRepository);
+        writeFile(remoteRepository, TEST_FILE_NAME, "Test Message");
+        remoteGit.add().addFilepattern(TEST_FILE_NAME).call();
+        RevCommit revCommit = remoteGit.commit().setMessage("Initial commit").call();
+        logger.info("Committed '{}' file to remote repository: {}", TEST_FILE_NAME, revCommit.getName());
+    }
```
[the remark this answers is not in the document]
Comment (olpaw) [reply tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/src/test/java/org_eclipse_jgit/org_eclipse_jgit/OrgEclipseJGitTest.java:59]: Ah I saw that you need this in `test()` because you are using `remoteGit.getRepository()`. Sorry for the confusion.

File: tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/src/test/java/org_eclipse_jgit/org_eclipse_jgit/OrgEclipseJGitTest.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                fail("Files not found in local repository");
+            }
+
+            localGit.getRepository().close();
+        } finally {
+            deleteDir(localRepositoryDir);
+        }
+    }
+
+    public void writeFile(Repository repository, String name, String data) throws IOException {
```
Comment (olpaw) [inline tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/src/test/java/org_eclipse_jgit/org_eclipse_jgit/OrgEclipseJGitTest.java:96]: Move this method up (right after `beforeAll()`).

**Comment by olpaw** [comment]
> The OrgEclipseJGitTest creates two github repositories in the tmp folder and deletes it at the end of the test execution.

I guess you mean: _creates two git repositories_ and not two github repositories, right?

File: tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/gradle.properties [hunk 5]
```diff
@@ -0,0 +1,2 @@
+library.version = 6.5.0.202303070854-r
```
Comment (olpaw) [inline tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/gradle.properties:1]: This is a very odd version string. Why does this not look like e.g. `6.5.0`? What is the reason for that?

File: tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/build.gradle (hunk 1 above)
Comment (msupic) [reply tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/build.gradle:26]: The `userCodeFilterPath` filters metadata entries by `typeReachable` class. The `extraFilterPath` filters by the metadata `name` class. Without `extraFilterPath` the following entry would end up in the generated metadata file
```
    {
        "name": "java.security.SecureRandomParameters",
        "condition": {
            "typeReachable": "org.eclipse.jgit.lib.Repository"
        }
    }
```
Since entries like that one are not needed, `extraFilterPath` is used.

File: tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/gradle.properties (hunk 2 above)
Comment (msupic) [reply tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/gradle.properties:2]: That file is generated by the `scaffold` task so if it is not needed, should be removed from that task.

File: tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/src/test/java/org_eclipse_jgit/org_eclipse_jgit/OrgEclipseJGitTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (msupic) [reply tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/src/test/java/org_eclipse_jgit/org_eclipse_jgit/OrgEclipseJGitTest.java:59]: Yes, it is used in the test

File: tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/gradle.properties (hunk 5 above)
Comment (msupic) [reply tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/gradle.properties:1]: That is version of a third party library and they create release names on that way:
`https://mvnrepository.com/artifact/org.eclipse.jgit/org.eclipse.jgit`

**Comment by msupic** [comment]
> > The OrgEclipseJGitTest creates two github repositories in the tmp folder and deletes it at the end of the test execution.
> 
> I guess you mean: _creates two git repositories_ and not two github repositories, right?

Modified, thanks

File: tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/src/test/java/org_eclipse_jgit/org_eclipse_jgit/OrgEclipseJGitTest.java (hunk 4 above)
Comment (msupic) [reply tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/src/test/java/org_eclipse_jgit/org_eclipse_jgit/OrgEclipseJGitTest.java]: Done

File: tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/build.gradle (hunk 1 above)
Comment (olpaw) [reply tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/build.gradle:26]: Thanks for the explanation!

File: tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/gradle.properties (hunk 5 above)
Comment (olpaw) [reply tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/gradle.properties:1]: I see. All the `org.eclipse.jgit` releases have such odd version names.

**Review by olpaw — APPROVED** [review]
